aptos-move/block-executor/src/txn_last_input_output.rs

```diff
@@ -211,33 +196,35 @@ impl<T: Transaction, O: TransactionOutput<Txn = T>, E: Debug + Send + Clone>
         )
     }
 
-    pub(crate) fn module_rw_intersection_ok(
+    pub(crate) fn check_execution_status_during_commit(
         &self,
+        txn_idx: TxnIndex,
     ) -> Result<(), PanicOr<IntentionalFallbackToSequential>> {
-        if self.module_read_write_intersection.load(Ordering::Acquire) {
-            Err(PanicOr::Or(
-                IntentionalFallbackToSequential::ModulePathReadWrite,
-            ))
+        if let Some(status) = self.outputs[txn_idx as usize].load_full() {
+            match &status.output_status {
+                ExecutionStatus::Success(_) | ExecutionStatus::SkipRest(_) => Ok(()),
+                // Transaction cannot be committed with below statuses, as:
+                // - Speculative error must have failed validation.
+                // - Execution w. delayed field code error propagates the error directly,
+                // does not finish execution. Similar for FatalVMError / abort.
+                ExecutionStatus::Abort(_) => {
+                    Err(code_invariant_error("Abort status cannot be committed").into())
+                },
+                ExecutionStatus::SpeculativeExecutionAbortError(_) => {
+                    Err(code_invariant_error("Speculative error status cannot be committed").into())
+                },
+                ExecutionStatus::DelayedFieldsCodeInvariantError(_) => Err(code_invariant_error(
+                    "Delayed field invariant error cannot be committed",
+                )
+                .into()),
+            }
         } else {
+            // TODO: check carefully if such a case is possible due to early halts.
+            debug!("Possible to not find committed output (halted?)");
```

Comment:
warn instead of debug, so we actually see it?

also do you want to quality TODO?

Reply:
Thanks, good point!
I will go over general block executor TODOs after all the refactors and classify them.